bot.py

```diff
@@ -641,9 +642,44 @@ def respond_to_ig_link_with_instafix(original_message, link):
 
 
 @bot.message_handler(regexp="http", func=lambda message: message.from_user.id in ALLOWED_USERS or message.chat.id in ALLOWED_CHATS)
-def handle_link(message):
-    if message.chat.id not in ALLOWED_CHATS:
-        bot.reply_to(message, "This site is not supported yet\.")
+def handle_unknown_link(message):
+    # Skip if already handled by other regexes
+    for regex in SITE_REGEXES.values():
+        if re.search(regex, message.text):
+            return
+
+    # Extract domain from URL
+    urls = re.findall(r'https?://[^\s]+', message.text)
+    for url in urls:
+        domain = re.findall(r'https?://([^/]+)/?', url)
+        if not domain:
+            continue
+        domain = domain[0]
+        # Skip if domain is Telegram or your bot
+        if "t.me" in domain or "telegram" in domain:
+            continue
+        # Check if it's a Mastodon instance (try v2, then v1)
+        try:
+            mastodon_instance = False
+            for api_version in ["v2", "v1"]:
+                resp = requests.get(f"https://{domain}/api/{api_version}/instance", timeout=3)
+                if resp.status_code == 200 and resp.headers.get("content-type", "").startswith("application/json"):
```

Comment:
better check should be done if possible - the url for mastodon api is very generic and other sites than mastodon could respond for this api call